packages/react/src/components/Forms/DatePicker/index.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { PropsWithChildren, useEffect, useRef, useState } from "react";
+import {
+  CalendarDate,
+  DateValue,
+  parseAbsoluteToLocal,
+  toCalendarDate,
+} from "@internationalized/date";
+import {
+  DatePickerStateOptions,
+  useDatePickerState,
+} from "@react-stately/datepicker";
+import { useDatePicker } from "@react-aria/datepicker";
+import classNames from "classnames";
+import { Button } from ":/components/Button";
+import { Field, FieldProps } from ":/components/Forms/Field";
+import { LabelledBox } from ":/components/Forms/LabelledBox";
+import { DateField } from ":/components/Forms/DatePicker/DateField";
+import { Calendar } from ":/components/Forms/DatePicker/Calendar";
+import { Popover } from ":/components/Popover";
+import { useCunningham } from ":/components/Provider";
+
+type DatePickerProps = PropsWithChildren &
+  FieldProps & {
+    label: string;
+    name?: string;
+    value?: null | Date | string;
+    defaultValue?: Date | string;
+    minValue?: Date | string;
+    maxValue?: Date | string;
+    onChange?: (value: string) => void | undefined;
+    disabled?: boolean;
+  };
+
+export const DatePicker = ({
+  label,
+  name,
+  disabled = false,
+  ...props
+}: DatePickerProps) => {
+  if (props.defaultValue && props.value) {
```

Comment:
Did you try to prevent this through typescript ?

Reply:
I have tried to be consistent with the `Select` component [l.339](https://github.com/openfun/cunningham/blob/0ad71a8ec1eae61c48a7c3b53e7d1ec057c88870/packages/react/src/components/Forms/Select/index.tsx#L339). In fact, the date picker component could have a default value and a controlled value. What do you think should I prevent this to be aligned with the select component ?

Reply:
Ok let's be consistent with Select for now.